python/cornserve/task_executors/geri/engine/core.py

```diff
@@ -390,3 +421,263 @@ def signal_handler(signum: int, frame) -> None:
         finally:
             if engine:
                 engine.shutdown()
+
+
+class BatchGeriEngine(Engine):
+    """Batched Geri core engine.
+
+    When content is generated, the engine sends response back to the router
+    in a batched, non-streaming manner.
+    """
+
+    def mode_specific_initializer(
+        self,
+        model: GeriModel,
+        geri_config: GeriConfig,
+    ) -> None:
+        """Initialize mode-specific (i.e., batch vs stream) components.
```

Comment:
This is typically done with a child class's `__init__`, and inside the child's `__init__`, it calls `super().__init__` at the right timing. Maybe this must be executed in the middle of the parent's `__init__` and that's why you did this? Anyways please see if just the conventional `__init__` works.